src/spartan/sumcheck.rs

```diff
@@ -92,6 +92,148 @@ impl<G: Group> SumcheckProof<G> {
       )
   }
 
+  #[tracing::instrument(skip_all, name = "Spartan2::sumcheck::prove_quad")]
+  // A fork of `prove_quad` with the 0th round unrolled from the rest of the 
+  // for loop. This allows us to pass in `W` and `X` as references instead of
+  // passing them in as a single `MultilinearPolynomial`, which would require
+  // an expensive concatenation. We defer the actual instantation of a 
+  // `MultilinearPolynomial` to the end of the 0th round.
+  pub fn prove_quad_unrolled<F>(
```

Comment:
Could clean this up by making compute_eval_points_quadratic take a param `impl Index<usize>`. Would still need the custom logic for binding unfortunately.